lib/launcher.rb

```diff
@@ -1,9 +1,21 @@
+require 'bundler'
 require 'proxy/log'
 require 'proxy/sd_notify'
 require 'proxy/settings'
 require 'proxy/signal_handler'
 require 'proxy/log_buffer/trace_decorator'
 require 'thread'
+require 'rack'
+require 'webrick'
```

Comment:
One thing in the suggested the original PR is waiting with requiring webrick vs. puma till the time we know which web server will be used (which is still at startup time). This way we could also avoid using `$HAS_PUMA` global variable.

Reply:
Yup done.

Reply:
Shouldn't this requiremens postponed to the time, where we check for the http_server_type?

Reply:
taking back, I've read 'puma' instead of 'rack', nm